apps/wallet/src/background/index.ts

```diff
@@ -24,4 +25,23 @@ Permissions.permissionReply.subscribe((permission) => {
 
 Keyring.on('lockedStatusUpdate', (isLocked: boolean) => {
     connections.notifyForLockedStatusUpdate(isLocked);
+    if (isLocked) {
+        Alarms.clearAlarm(LOCK_ALARM_NAME);
+    } else if (connections.totalUiConnections === 0) {
+        Alarms.setLockAlarm();
+    }
+});
+
+connections.on('totalUiChanged', (ui) => {
```

Comment:
How sure are we that these events fire? In general, I'm not super stoked on this approach to setting the timer for locking, it feels like it's too easy to forget to set the timer in certain circumstances. I'd rather our approach here be to bake this into the keyring itself, so that it _always_ sets a timer (and cannot _not_  set a timer). Any interaction with the keyring would just reset the lock timer. The UI can send a keepalive to keep the keyring unlocked as well.

The upside to that approach is that there's never a case where the keyring doesn't lock due to a bug on the event side here, the only thing that could happen is that we accidentally lock instead, which is okay.

Reply:
Sounds a good idea, I mostly did it this way to keep it simple and quick